debezium-core/src/main/java/io/debezium/relational/Tables.java

```diff
@@ -182,8 +182,14 @@ public int size() {
     public Table overwriteTable(TableId tableId, List<Column> columnDefs, List<String> primaryKeyColumnNames,
                                 String defaultCharsetName) {
         return lock.write(() -> {
-            TableImpl updated = new TableImpl(tableId, columnDefs, primaryKeyColumnNames, defaultCharsetName);
```

Comment:
Nice, this was bothering me, too :)

Reply:
It actually causes a bug with the rest of the implementation because the TableEditor shifts the positions of the columns (something I found unexpected).

Reply:
Interesting; so are you reverting this one then?

What's unclear to me is why we need that explicit position field in `Column` in the first place; I'd assume we'd always be able to derive that info from the list index. But seems rather risky to change right now and it's definitely a separate discussion.

Reply:
Sorry, I meant that I needed to make that change to solve the bug. 

I prefer using the position as it seems to prevent issues around excluding columns (that's what my past three PRs have been about 😅 ). It just feels a little safer to me than the raw index.